collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java

```diff
@@ -231,31 +220,163 @@ private boolean uploadOneSubmission(String id, String instanceFilePath, String j
+         *  Media files are uploaded to Google Drive of user
+         *  All media files are currently saved under folder "odk-uploads"
+         */
+
+        // if we have any media files to upload,
+        // get the folder or create a new one
+        // then upload the media files
         if (photosToUpload.size() > 0) {
-            // First set up a picasa album to upload to:
-            // maybe we should move this, because if we don't have any
-            // photos we don't care...
-            AlbumEntry albumToUse;
-            try {
-                albumToUse = getOrCreatePicasaAlbum(client, jrFormId);
-            } catch (IOException e) {
-                GoogleAuthUtil.invalidateToken(Collect.getInstance(), token);
-                mResults.put(id, picasa_fail + e.getMessage());
-                return false;
-            }
 
-            try {
-                uploadPhotosToPicasa(photosToUpload, uploadedPhotos, client, albumToUse,
-                        instanceFile);
-            } catch (IOException e1) {
-                mResults.put(id, picasa_fail + e1.getMessage());
-                return false;
+            for (String key : photosToUpload.keySet()) {
+                String filename = instanceFile.getParentFile() + "/" + photosToUpload.get(key);
+                File toUpload = new File(filename);
+
+                // first check the local content provider
+                // to see if this photo still exists at the location or not
+                String selection = MediaStore.Images.Media.DATA + "=?";
+                String[] selectionArgs = {
+                        filename
+                };
+                Cursor c = Collect.getInstance().getContentResolver()
+                        .query(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, null, selection
+                                , selectionArgs, null);
+                if (c.getCount() != 1) {
+                    c.close();
+                    try {
+                        throw new FileNotFoundException(drive_fail + Collect.getInstance()
+                                .getString(R.string.media_upload_error, filename));
+                    } catch (FileNotFoundException e) {
+                        e.printStackTrace();
+                    }
+                }
+                c.close();
+
+                // try to get root folder "Open Data Kit"
+                ArrayList<com.google.api.services.drive.model.File> files = new ArrayList<>();
+                try {
+                    getFilesFromDrive(GOOGLE_DRIVE_ROOT_FOLDER, files, null);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    mResults.put(id, form_fail + e.getMessage());
+                    return false;
+                }
+
+                com.google.api.services.drive.model.File folder = null;
+
+                for (com.google.api.services.drive.model.File file : files) {
+                    if (folder == null) {
```

Comment:
The reason behind doing this looping was to verify that only one instance of the folder exists. If multiple folders are found then it throws an error message "2 folders found". Is this the correct approach?